packages/application-config/src/process-config.ts

```diff
@@ -69,18 +81,45 @@ const processConfig = ({
       ),
     `Invalid MC API URL: "${appConfig.mcApiUrl}"`
   );
-  // TODO: make the `id` required once we release the new model.
-  const applicationId =
-    appConfig.id && appConfig.entryPointUriPath
-      ? `${appConfig.id}:${appConfig.entryPointUriPath}`
-      : undefined;
+
+  // The real application ID is only used in production.
+  // In development, we prefix the entry point with the "__local" prefix.
+  // This is important to determine to which URL the MC should redirect to
+  // after successful login.
+  let applicationId: string | undefined = isOidcForDevelopmentEnabled
+    ? `__local:${appConfig.entryPointUriPath}`
+    : undefined;
+  if (isProd) {
+    // TODO: decide if we do require the application ID or not.
+    if (appConfig.env.production.applicationId) {
+      applicationId = `${appConfig.env.production.applicationId}:${appConfig.entryPointUriPath}`;
+    } else {
+      // As long as we don't require the application ID in production, we should
+      // fall back to unset the value.
+      applicationId = undefined;
+    }
+  }
 
   cachedConfig = {
     env: {
       ...omitEmpty(additionalAppEnv),
-      ...(applicationId ? { applicationId } : {}),
+      // TODO: how else should we provide the app identifier?
+      applicationId,
       applicationName: appConfig.name,
       entryPointUriPath: appConfig.entryPointUriPath,
+      ...(!isProd && isOidcForDevelopmentEnabled
+        ? {
+            __DEVELOPMENT__: omitEmpty({
+              authorizeUrl: `${mcApiUrl.protocol}//${mcApiUrl.host.replace(
```

Comment:
Should we explicitly define `authorizeUrl` or is there a benefit from inferring it.

Reply:
We should infer it, because the environment (staging, production, etc) has to be the same with the APIs that the application uses.